doc/source/data/api/dataset.rst

```diff
@@ -149,3 +149,17 @@ Serialization
    Dataset.has_serializable_lineage
    Dataset.serialize_lineage
    Dataset.deserialize_lineage
+
+
```

Comment:
We need these to be explicitly in a TOC somewhere. We can argue about the exact position after fixing the build.